lib/storage/src/content_manager/toc.rs

```diff
@@ -190,7 +190,8 @@ impl TableOfContent {
         let CreateCollection {
             vector_size,
             distance,
-            shard_number,
+            // Can be safely ignored as `collection_shard_distribution` already has more accurate information
+            shard_number: _shard_number,
```

Comment:
What about removing the `shard_number` argument completely?

Reply:
This other suggestion is to add an assert making sure that the `shard_number` is reflected in the `shard_distribution` when it is defined.
So we ignore it, but we have a clear documentation and minimal harness for future invalid state.

Reply:
After quick discussion on a call we came to conclusion that we should just assert it for consistency.